ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs

```diff
@@ -17,6 +17,121 @@
 {-# LANGUAGE TupleSections              #-}
 {-# LANGUAGE TypeApplications           #-}
 
+{-|
+  = Snapshot managing and contents
+
+  Snapshotting a ledger state means saving a copy of the in-memory part of the
+  ledger state serialized as a file in the disk, as well as flushing differences
+  between the last snapshotted ledger state and the one that we are snapshotting
+  now onto the BackingStore. While the coupled implementation is in place, the
+  serialized file might also contain a serialized version of the in-memory UTxO
+  map carried by the legacy ledger states, as explained below.
+
+  == Startup
+
+  On startup, the node will:
+
+  1. Find the latest snapshot which will be a directory identified by the slot
+     number of the snapshot:
+
+        > <cardano-node data dir>
+        > ├── volatile
+        > ├── immutable
+        > └── ledger
+        >     ├── <slotNumber1>
+        >     │   ├── tables
+        >     │   └── state
+        >     ├── <slotNumber2>
+        >     │   ├── tables
+        >     │   └── state
+        >     └── <slotNumber3>
+        >         ├── tables
+        >         └── state
+
+  2. Depending on the snapshots found, there are two paths:
+
+       - If there is no snapshot to load, create a new @BackingStore@ with the
+         contents of the Genesis UTxO and finish.
+
+       - If there is a snapshot found, then deserialize the @state@ file, which
+         contains either:
+
+            - If the node was running __with the legacy database enabled__, it
+              contains a full LedgerState snapshot in the same style as before
+              the UTxO HD implementation (i.e. in-memory part of the LedgerState
+              (equal among implementations) + UTxO map).
+
+            - Otherwise, it contains a serialization of the in-memory part of the
+              LedgerState, without a UTxO map.
+
+        In both cases, the snapshot will be deserialized as @ExtLedgerState blk
+        EmptyMK@. In case we found an snapshot, we will overwrite the
+        @BackingStore@ tables with the @tables@ file from said snapshot.
```

Comment:
Maybe we should mention briefly the abstraction that the backing store represents, and what is its relation w.r.t. the tables it contains. Also shouldn't we describe in the beginning how we initialize this backing store we're overwriting in this step? Is it an empty one? Or one that is in whatever state it was left when the node was shut down? 

Another question is: if we're overwriting the tables of the backing store, would it make sense to always start with an initialized backing store? (eg an empty one)

Reply:
This one
> Or one that is in whatever state it was left when the node was shut down?
Not sure if we should mention that here though. I can see benefits on doing it and also on not doing it but instead adding that to the BackingStore definition